File: arrhash/sudoku.js
let board = [
  ["1", "2", ".", ".", "3", ".", ".", ".", "."],
  ["4", ".", ".", "5", ".", ".", ".", ".", "."],
  [".", "9", "8", ".", ".", ".", ".", ".", "3"],
  ["5", ".", ".", ".", "6", ".", ".", ".", "4"],
  [".", ".", ".", "8", ".", "3", ".", ".", "5"],
  ["7", ".", ".", ".", "2", ".", ".", ".", "6"],
  [".", ".", ".", ".", ".", ".", "2", ".", "."],
  [".", ".", ".", "4", "1", "9", ".", ".", "8"],
  [".", ".", ".", ".", "8", ".", ".", "7", "9"],
];

function isValidRowsColumns(board) {
  const hm_columns = Array.from({ length: 9 }, () => []);
  for (let row of board) {
    const row_items = [];
    for (let i = 0; i < row.length; i++) {
      if (row[i] == ".") {
        continue;
      }
      if (row_items.includes(row[i])) {
        return false;
      }
      row_items.push(row[i]);
      if (hm_columns[i].includes(row[i])) {
        return false;
      }
      hm_columns[i].push(row[i]);
    }
  }
  return true;
}

function isValidSquare(board) {
  const hm_squares = Array.from({ length: 9 }, () => []);
  for (let i = 0; i < board.length; i++) {
    for (let j = 0; j < board[i].length; j++) {
      const squareIndex = Math.floor(i / 3) * 3 + Math.floor(j / 3);
      if (board[i][j] == ".") {
        continue;
      }
      if (hm_squares[squareIndex].includes(board[i][j])) {
        return false;
      }
      hm_squares[squareIndex].push(board[i][j]);
    }
  }
  return true;
}

const result = isValidRowsColumns(board) && isValidSquare(board);
console.log(result);

document.getElementById("result").textContent = "Resultado: " + result;

File: arrhash/sudoku.py
board = [["1","2",".",".","3",".",".",".","."],
 ["4",".",".","5",".",".",".",".","."],
 [".","9","1",".",".",".",".",".","3"],
 ["5",".",".",".","6",".",".",".","4"],
 [".",".",".","8",".","3",".",".","5"],
 ["7",".",".",".","2",".",".",".","6"],
 [".",".",".",".",".",".","2",".","."],
 [".",".",".","4","1","9",".",".","8"],
 [".",".",".",".","8",".",".","7","9"]]

def isValidRowsColumns(board):
  hm_columns = {i: [] for i in range(0, 9)}
  for row in board:
    row_items = []
    for i, n in enumerate(row):
      if n == ".":
        continue
      if n in row_items:
        return False
      row_items.append(n)
      if n in hm_columns[i]:
        return False
      hm_columns[i].append(n)
  return True

def isValidSquare(board):
  hm_squares = {i: [] for i in range(0, 9)}
  for i, row in enumerate(board):
    for j, n in enumerate(row):
      if n == ".":
        continue
      if n in hm_squares[i // 3 * 3 + j // 3]:
        return False
      hm_squares[i // 3 * 3 + j // 3].append(n)
  return True

result = isValidRowsColumns(board) and isValidSquare(board)
print(result)